cmd/build.go

```diff
@@ -1,32 +1,24 @@
 package cmd
 
 import (
-	"errors"
 	"fmt"
 	"io"
-	"io/fs"
 	"os"
 	"path/filepath"
+	"slices"
 	"strings"
 
 	"github.com/linux-immutability-tools/EtcBuilder/core"
 	"github.com/linux-immutability-tools/EtcBuilder/settings"
-	"golang.org/x/exp/slices"
-
 	"github.com/spf13/cobra"
 )
 
-type NoMatchError struct{}
-
-func (m *NoMatchError) Error() string {
-	return "Specified files are not the same"
+type FileHandler struct {
+	IsFileSupported func(path string) bool
+	Handle          func(relativeFilePath, oldSysDir, newSysDir, oldUserDir, newUserDir string) error
 }
```

Comment:
Structs (including functions that belong to them like the `Error` one here) should be above all other functions